orttraining/orttraining/core/agent/training_agent.cc

```diff
@@ -1,6 +1,8 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+#if defined(ENABLE_TRAINING) && !defined(ORT_MINIMAL_BUILD)
```

Comment:
I think this can be done in the cmake file, just exclude the whole folder from compiling. 
This might already been done. 

I wonder if we even include orttraining folder under the Minimal build.

Reply:
same comment applies to the .h file.

Reply:
cmake/onnxruntime_training.cmake doesnt seem to have any toggle for that

Do we doo training at all with minimal build? 

what are the scenarios for minimal build?

Reply:
We have this in the main Cmakelists.txt:

```cmake
if (onnxruntime_ENABLE_TRAINING)
  include(onnxruntime_training.cmake)
  if (onnxruntime_ENABLE_TRAINING_E2E_TESTS)
    include(onnxruntime_training_e2e_tests.cmake)
  endif()
endif()
```